File: UA1/tareas/Tarea1.java

/*
    CFGS Desarrollo Aplicaciones Multiplataforma
    Programación Servicios y Procesos
    CES Lope de Vega - Córdoba (Spain)
    Curso 2022/2023
*/

/*
    Tarea 1 - Programación de Procesos en C
    Criterios a), f), g), h)
    Crea un programa en C que cree un proceso (existirán en realidad dos procesos, uno padre y el otro hijo). 
    El programa principal pedirá al usuario una variable y el proceso padre restará 5 a dicha variable, mientras que el proceso hijo le sumará 4. 
    Muestra todos los valores por pantalla.
*/
//import java.*;
 
//package ua1;

// https://medium.com/@pelensky/java-tdd-with-junit-without-using-an-ide-cd24d38adff

#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>

int main() {
   pid_t proceso, proceso2;//Se crean las derivadas
   int num;
   
   printf("Introduce un numero");//se le pide un numero al usuario
   scanf("%d",&num);
   
   proceso=fork();
   
   if(proceso==-1){//Si ocurre un error el programa termina
       printf("ERROR, No se ha creado al proceso hijo");
       exit(-1);  
   }
   if(proceso==0){//se pone el calculo a realizar en el proceso hijo
       num=num+4;

   }
   else{//se pone el calculo a realizar en el proceso padre
       wait(NULL);
       num=num-5;
   }
  
printf("Resultado: %d\n",num);//Se imprimen los resultados en pantalla
   exit(0);
}

File: UA1/tareas/Tarea1.c
#include <stdlib.h>
#include <unistd.h>
#include <stdio.h>
void main() {
  pid_t proceso1;
  
  int numero;//Numero que pediremos al usuario
  
  //Pedimos al usuario el numero para operar con él
  printf("Introduzca un numero");
  scanf("%d", &numero);
  
  //Creamos el proceso hijo
  proceso1 = fork();
  
  //Comprobamos que el proceso se haya creado correctamente
  if(proceso1 == -1){
      printf("Error al crear el proceso");
  }
  
  //Acciones del proceso hijo
  else if(proceso1 == 0){
      numero = numero + 4;
  }
  
  //Acciones del proceso padre
  else{
      wait(NULL); //Espera a que la ejecución el proceso hijo finalice
      numero = numero - 5;
  }
    //Muestra los resultados por pantalla
      printf("Variable:%d\n",numero);
}
